homeassistant/components/awair/sensor.py

```diff
@@ -1,248 +1,241 @@
-    },
-    "score": {
-        "device_class": DEVICE_CLASS_SCORE,
-        "unit_of_measurement": UNIT_PERCENTAGE,
-        "icon": "mdi:percent",
-    },
-}
-
-AWAIR_QUOTA = 300
-
-# This is the minimum time between throttled update calls.
-# Don't bother asking us for state more often than that.
-SCAN_INTERVAL = timedelta(minutes=5)
-
-AWAIR_DEVICE_SCHEMA = vol.Schema({vol.Required(CONF_UUID): cv.string})
-
-PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_ACCESS_TOKEN): cv.string,
-        vol.Optional(CONF_DEVICES): vol.All(cv.ensure_list, [AWAIR_DEVICE_SCHEMA]),
-    }
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    API_DUST,
+    API_PM25,
+    API_SCORE,
+    API_TEMP,
+    API_VOC,
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_UNIQUE_ID,
+    ATTR_UNIT,
+    ATTRIBUTION,
+    DOMAIN,
+    DUST_ALIASES,
+    SENSOR_TYPES,
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_ACCESS_TOKEN): cv.string}, extra=vol.ALLOW_EXTRA,
 )
 
 
-# Awair *heavily* throttles calls that get user information,
-# and calls that get the list of user-owned devices - they
-# allow 30 per DAY. So, we permit a user to provide a static
-# list of devices, and they may provide the same set of information
-# that the devices() call would return. However, the only thing
-# used at this time is the `uuid` value.
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Connect to the Awair API and find devices."""
-
-    token = config[CONF_ACCESS_TOKEN]
-    client = AwairClient(token, session=async_get_clientsession(hass))
-
-    try:
-        all_devices = []
-        devices = config.get(CONF_DEVICES, await client.devices())
-
-        # Try to throttle dynamically based on quota and number of devices.
-        throttle_minutes = math.ceil(60 / ((AWAIR_QUOTA / len(devices)) / 24))
-        throttle = timedelta(minutes=throttle_minutes)
-
-        for device in devices:
-            _LOGGER.debug("Found awair device: %s", device)
-            awair_data = AwairData(client, device[CONF_UUID], throttle)
-            await awair_data.async_update()
-            for sensor in SENSOR_TYPES:
-                if sensor in awair_data.data:
-                    awair_sensor = AwairSensor(awair_data, device, sensor, throttle)
-                    all_devices.append(awair_sensor)
-
-        async_add_entities(all_devices, True)
-        return
-    except AwairClient.AuthError:
-        _LOGGER.error("Awair API access_token invalid")
-    except AwairClient.RatelimitError:
-        _LOGGER.error("Awair API ratelimit exceeded.")
-    except (
-        AwairClient.QueryError,
-        AwairClient.NotFoundError,
-        AwairClient.GenericError,
-    ) as error:
-        _LOGGER.error("Unexpected Awair API error: %s", error)
-
-    raise PlatformNotReady
+    """Import Awair configuration from YAML."""
+    hass.async_create_task(
```

Comment:
Should we log a warning about the deprecation here and ask users to remove the config yaml section?

Reply:
Not a bad idea; I added it in 32fffdfafa9b1771095fbd0f63b0ce02be88cd04:

```
~/projects/home-assistant ahayworth-awair-revamp* ⇡ 10s
venv ❯ git show
commit 32fffdfafa9b1771095fbd0f63b0ce02be88cd04 (HEAD -> ahayworth-awair-revamp)
Author: Andrew Hayworth <ahayworth@gmail.com>
Date:   Sun Jun 21 12:14:08 2020 -0500

    Add warning about deprecated platform config

diff --git a/homeassistant/components/awair/sensor.py b/homeassistant/components/awair/sensor.py
index fc1e1869b1..e4e2f3fbbd 100644
--- a/homeassistant/components/awair/sensor.py
+++ b/homeassistant/components/awair/sensor.py
@@ -27,6 +27,7 @@ from .const import (
     ATTRIBUTION,
     DOMAIN,
     DUST_ALIASES,
+    LOGGER,
     SENSOR_TYPES,
 )

@@ -37,6 +38,9 @@ PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(

 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Import Awair configuration from YAML."""
+    LOGGER.warning(
+        "Loading Awair via platform setup is deprecated. Please remove it from your configuration."
+    )
     hass.async_create_task(
         hass.config_entries.flow.async_init(
             DOMAIN, context={"source": SOURCE_IMPORT}, data=config,

~/projects/home-assistant ahayworth-awair-revamp* ⇡
```

Feel free to adjust the language of the warning if you'd like, though - doesn't seem to be much of a standard to compare against. I don't feel strongly about how we say this to users, either.